src/history/factory.ts

```diff
@@ -23,20 +23,22 @@ export function buildRevisionLog(
       revision.setChanges(changes);
     },
     (revision: Revision) => revertChanges([revision]),
-    (id: UID) => new Revision(id, "empty", [], []),
+    (id: UID) => new Revision(id, "empty", [], undefined, []),
     {
       with: (revision: Revision) => (toTransform: Revision) => {
         return new Revision(
           toTransform.id,
           toTransform.clientId,
-          transformAll(toTransform.commands, revision.commands)
+          transformAll(toTransform.commands, revision.commands),
+          toTransform.rootCommand
```

Comment:
The root command itself yeah, but we can still repeat it since we adapt the sheetId/target to the current sheet/current selection.

Except for `DUPLICATE_SHEET`, but maybe it should duplicate the active sheet, and not re-duplicate the sheet previously duplicated.

Do you see a problem with that ?

Reply:
> Except for DUPLICATE_SHEET, but maybe it should duplicate the active sheet, and not re-duplicate the sheet previously duplicated

Or simply not repeatable ? It looks like it's not repeatable on GSheet

Reply:
Probably easier yeah